### PR TITLE
🐛 correct the agent deployment name in different install mode

### PR DESCRIPTION
**Pull request by elgnay**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/operator/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go [hunk 1]
```diff
@@ -181,7 +181,11 @@ func (k *bootstrapController) sync(ctx context.Context, controllerContext factor
 
 func (k *bootstrapController) processRebootstrap(ctx context.Context, agentNamespace string, klusterlet *operatorapiv1.Klusterlet,
 	recorder events.Recorder, requeueFunc func(time.Duration)) error {
-	deploymentName := fmt.Sprintf("%s-registration-agent", klusterlet.Name)
+	deploymentName := fmt.Sprintf("%s-agent", klusterlet.Name)
+	if len(klusterlet.Spec.DeployOption.Mode) == 0 || klusterlet.Spec.DeployOption.Mode == operatorapiv1.InstallModeDefault ||
```
Comment (qiujian16) [inline pkg/operator/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go:185]: I think it is not Singleton. We have a helper func to check it

Comment (elgnay) [reply pkg/operator/operators/klusterlet/controllers/bootstrapcontroller/bootstrapcontroller.go]: Updated as suggested.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**